middle_end/flambda2/types/flambda2_types.mli

```diff
@@ -578,15 +578,15 @@ val prove_variant_like :
   Typing_env.t -> t -> variant_like_proof proof_of_property
 
 (** If [ty] is known to represent a boxed number or a tagged integer,
-    [prove_is_a_boxed_number env ty] is [Proved kind]. [kind] is the kind of the
-    unboxed number.
+    [prove_is_a_boxed_number env ty] is [Proved (alloc_mode,kind,contents_ty)].
+    [kind] is the kind of the unboxed number.
 
     If [ty] is known to represent something of kind value that is not a number
     [prove_is_a_boxed_number env ty] is [Invalid].
 
     Otherwise it is [Unknown] or [Wrong_kind] when [ty] is not of kind value. *)
```

Comment:
Remind me at some point to make a pass on this file to fix the comments (they haven't been updated in the Provers overhaul PR)

Reply:
Have made a note.